src/types.rs

```diff
@@ -167,3 +171,35 @@ pub struct GetNetworkInfoResult {
     pub local_addresses: Vec<GetNetworkInfoResultAddress>,
     pub warnings: String,
 }
+
+#[derive(Serialize, Clone, Eq, PartialEq, Debug, Hash)]
+pub struct DescriptorChecksum(pub String);
+
+impl FromStr for DescriptorChecksum {
+    type Err = BwtError;
+    fn from_str(s: &str) -> Result<DescriptorChecksum, Self::Err> {
+        Ok(DescriptorChecksum(s.to_string()))
+    }
+}
+
+#[derive(Serialize, Clone, Eq, PartialEq, Debug, Hash)]
+pub struct Descriptor(pub String);
```

Comment:
Perhaps this should be an enum or "ranged" and "unranged" descriptors?

We could distinguish by matching `*` inside the string?

Reply:
I think this could be a boolean field. `getdescriptorinfo` returns an `isrange` field that we could use instead of matching `*`.

Reply:
I agree that could work, but I'm going to leave this work to a future PR

Reply:
This could be `From<&str>` or `From<String>` since the conversion is not fallible. (they return `T` directly and not a `Result<T>` like `FromStr`.)

Though, if its just wrapping a `String`, this could also simply be constructed using `DescriptorChecksum(cs_string)` instead of `DescriptorChecksum::from(cs_string)`.

Reply:
Actually, it seems like you are in fact constructing them with `DescriptorChecksum(cs_string)`, it seems like the conversion method here could simply be removed.